python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_lognormal.py

```diff
@@ -0,0 +1,293 @@
+#    using the polynomials orthonormal with respect to the LogNormal distribution,
+#  - **Case 2:** we show how to obtain a higher-quality surrogate model
+#    using another polynomial family: the polynomials orthonormal
+#    with respect to the :class:`~openturns.Normal` distribution.
+
+# %%
+import openturns as ot
+import openturns.viewer as otv
+from math import sqrt
+# sphinx_gallery_thumbnail_number = 7
+
+# %%
+# Introduction: orthonormal polynomial family w.r.t. the LogNormal distribution
+# -----------------------------------------------------------------------------
+#
+# In this section, we build the family of polynomials orthonormal with respect to the LogNormal distribution
+# using the :class:`~openturns.AdaptiveStieltjesAlgorithm`.
+#
+# We build the polynomials up to degree 40; they are defined by their three-term recurrence coefficients.
+dist_X = ot.LogNormal()
+lower_Bound = dist_X.computeQuantile(0.01, False)
+upper_Bound = dist_X.computeQuantile(0.01, True)
+degreeMax = 30
+
+as_Algo = ot.AdaptiveStieltjesAlgorithm(dist_X)
+sample_Coeff = ot.Sample(0, 3)
+for i in range(degreeMax):
+    coeff = as_Algo.getRecurrenceCoefficients(i)
+    sample_Coeff.add(coeff)
+
+# %%
+# We build a set of polynomials with increasing degrees.
+# For each polynomial, we display its value at :math:`x = 0`.
+#
+# We anticipate that the vicinity of :math:`x = 0` will cause difficulties
+# for the surrogate model of :math:`\model`, since
+# :math:`\lim_{x \rightarrow 0} \model(x) = +\infty`.
+# Given that the polynomials take *reasonable values* in the neighborhood of :math:`x = 0`,
+# we expect the polynomial surrogate model to require
+# very large linear coefficients in order to reproduce high values near :math:`x = 0`,
+# which may lead to numerical cancellation issues.
+#
+# We explore polynomial degrees up to 40.
+deg_list = [5, 10, 20, 30, 40]
+g = ot.Graph('Polynomials orthonormal  with respect to the LogNormal distribution', 'x', 'y', True)
+leg = ot.Description(0)
+for indice, deg in enumerate(deg_list):
+    coeff_Poly_Deg = sample_Coeff[0:deg]
+    poly_Deg = ot.OrthogonalUniVariatePolynomial(coeff_Poly_Deg)
+    g.add(poly_Deg.draw(lower_Bound[0], upper_Bound[0], 1024))
+    leg.add(r'$d = $' + str(deg))
+    print('degree = ', deg, 'polynomial(0) = ', poly_Deg(0))
+
+g.setLegends(leg)
+g.setLegendPosition('topleft')
+view = otv.View(g)
+
+# %%
+# We observe that the polynomials up to degree 40:
+#
+# - are computed in a stable manner, as indicated by their smooth graphs,
+# - take reasonable values on the interval :math:`[0, 25]`.
+
+# %%
+# Case 1: PCE using orthonormal polynomial family w.r.t. the LogNormal distribution
+# ---------------------------------------------------------------------------------
+#
+# We define :math:`\model`, the input random variable :math:`X` and the output random variable :math:`Y = \model(X)`.
+input_Dim = dist_X.getDimension()
```

Comment:
This is not used _at all_ in the script.
```suggestion
```

Reply:
yes!